### PR TITLE
Make controller aware of clouds

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/Cloud.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.api.integration.zone;
+
+/**
+ * Known clouds.
+ *
+ * @author mpolden
+ */
+public enum Cloud {
```
Comment (freva) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/Cloud.java]: Since this is open source, maybe this should be a wrapped string like `RegionName`?

Comment (mpolden) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/Cloud.java]: I'll give it a try and see how it fits with the abstraction I chose for per-cloud OS upgrades.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java [hunk 2]
```diff
@@ -161,7 +161,7 @@ private void compare(HttpResponse response, String expected) {
 
     private void assertFile(HttpResponse response, String resourceName) {
         try {
-            Path path = Paths.get(getClass().getClassLoader().getResource(resourceName).getPath());
```
Comment (andreer) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java]: Just curious, why this change?

**Review by andreer — APPROVED** [review]

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java (hunk 2 above)
Comment (mpolden) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java]: IntelliJ sometimes gets confused which class path to use, which causes tests that use this method to fail with NPE. I think it happens after I install the module locally with `mvn install`, which I sometimes do to check for breaking API between this and internal repo.
